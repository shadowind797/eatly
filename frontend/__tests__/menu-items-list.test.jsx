import React from 'react';
import {render, fireEvent, waitFor} from '@testing-library/react';
import {MemoryRouter as Router} from "react-router-dom";
import MenuItemsList from '../src/components/Menu/Items.jsx';

describe("Menu-Items-List component", () => {
    it('renders without crashing', () => {
        render(<Router><MenuItemsList type="items" items={[]}/></Router>);
    });

    it('matches the snapshot', () => {
        const {asFragment} = render(<Router><MenuItemsList type="items" items={[{"not_found": "no items"}]}/></Router>);
        expect(asFragment()).toMatchSnapshot();
    });
})
