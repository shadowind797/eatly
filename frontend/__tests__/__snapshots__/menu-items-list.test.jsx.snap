// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu-Items-List component matches the snapshot when dishes found 1`] = `
<DocumentFragment>
  <div
    class="container"
    data-testid="filtered-items"
    id="items-list"
  >
    <div
      class="dish"
    >
      <img
        alt=""
        class="photo"
        src="undefinedhttp://127.0.0.1:8000/media/img/food/ChikenHell.png"
      />
      <img
        alt=""
        class="like"
        src="test-file-stub"
      />
      <div
        class="info"
      >
        <div
          class="item-cat"
        >
          <div
            style="background-color: rgb(247, 237, 208); color: rgb(218, 163, 26);"
          />
          <h4>
            Cheeseburger
          </h4>
        </div>
        <div
          class="time-rating"
        >
          <p
            class="rest"
          />
          <div>
            <p
              data-testid="item-rate"
            >
              4.9
            </p>
            <img
              alt=""
              src="test-file-stub"
            />
          </div>
        </div>
        <div
          class="price-add"
        >
          <h5
            data-testid="item-cost"
          >
            14
            <span>
              .99
            </span>
          </h5>
          <button
            data-testid="add-to-cart-button"
          >
            <img
              alt=""
              src="test-file-stub"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="dish"
    >
      <img
        alt=""
        class="photo"
        src="undefinedhttp://127.0.0.1:8000/media/img/food/ChikenHell.png"
      />
      <img
        alt=""
        class="like"
        src="test-file-stub"
      />
      <div
        class="info"
      >
        <div
          class="item-cat"
        >
          <div
            style="background-color: rgba(238, 0, 255, 0.32); color: rgb(255, 0, 242);"
          >
            Asian
          </div>
          <h4>
            Chicken Hell
          </h4>
        </div>
        <div
          class="time-rating"
        >
          <p
            class="rest"
          />
          <div>
            <p
              data-testid="item-rate"
            >
              4.8
            </p>
            <img
              alt=""
              src="test-file-stub"
            />
          </div>
        </div>
        <div
          class="price-add"
        >
          <h5
            data-testid="item-cost"
          >
            12
            <span>
              .99
            </span>
          </h5>
          <button
            data-testid="add-to-cart-button"
          >
            <img
              alt=""
              src="test-file-stub"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Menu-Items-List component matches the snapshot when no items found 1`] = `
<DocumentFragment>
  <div
    id="no-items"
  >
    <img
      alt=""
      src="test-file-stub"
      style="width: 200px;"
    />
    <h2>
      No items found
    </h2>
    <p>
      Please, try searching for something else
    </p>
  </div>
</DocumentFragment>
`;

exports[`Menu-Items-List component matches the snapshot when restaurants found 1`] = `
<DocumentFragment>
  <div
    class="container"
    id="items-list"
  >
    <div
      class="rest-card"
    >
      <img
        alt=""
        class="rest-img"
        src="undefinedhttp://127.0.0.1:8000/media/img/rests/mcdonalds.jpg"
      />
      <div
        class="info"
      >
        <div
          class="cat"
          style="background-color: rgb(247, 237, 208); color: rgb(218, 163, 26);"
        >
          Supreme
        </div>
        <div
          class="text"
        >
          <h4>
            The Burger King
          </h4>
          <div
            class="widgets"
          >
            <div
              class="time-rating"
            >
              <p>
                 dishes
              </p>
              <div>
                <img
                  alt=""
                  src="test-file-stub"
                />
                <p>
                  4.9
                </p>
              </div>
            </div>
            <button>
              <img
                alt=""
                src="test-file-stub"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
